rust/src/pgsql/pgsql.rs

```diff
@@ -343,14 +360,20 @@ impl PgsqlState {
                 Ok((rem, request)) => {
                     sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
+                    let mut temp_state = PgsqlStateProgress::IdleState;
                     if let Some(state) = PgsqlState::request_next_state(&request) {
                         self.state_progress = state;
+                        temp_state = state;
                     };
-                    let tx_completed = self.is_tx_completed();
+                    let tx_completed = self.is_tx_completed(Direction::ToServer);
                     if let Some(tx) = self.find_or_create_tx() {
                         tx.request = Some(request);
                         if tx_completed {
-                            tx.tx_state = PgsqlTransactionState::ResponseDone;
+                            if temp_state == PgsqlStateProgress::ConnectionTerminated || temp_state == PgsqlStateProgress::CancelRequestReceived { 
+                                tx.tx_state = PgsqlTransactionState::ResponseDone;
```

Comment:
with the commit msg staying "track per direction" I would expect a per direction variable, just like with htp. How is this a per direction tracker?

Reply:
I'm probably misinterpreting what the tracking per direction meant. My thought was that if doing this per request or response, it would already be per-direction. this specific case is for situations when there's no answer from the backend/server, so these states mean the whole transaction is over - that's why I used it this way.

Reply:
Would the `htp` example be this? https://github.com/OISF/libhtp/blob/0.5.x/htp/htp_transaction.h#L75-L91

Reply:
yeah, see https://github.com/OISF/libhtp/blob/0.5.x/htp/htp.h#L532